File: 251229/사는 지역/where-live.js
const fs = require('fs');

class Person {
    constructor(name, bunji, region) {
        this.name = name;
        this.bunji = bunji;
        this.region = region;
    }
}

const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const people = [];
const names = []
for (let i = 1; i <= n; i++) {
    const [name, bunji, region] = input[i].split(' ');
    people.push(new Person(name, bunji, region));
    names.push(name);
}

names.sort();
const idx = names.length - 1;

for (let i = 0; i < n; i++) {
    if (people[idx].name == people[i].name) {
        console.log(`name ${ans.name}`);
        console.log(`addr ${ans.bunji}`);
        console.log(`city ${ans.region}`);
        break;
    }
}

File: 251229/사는 지역/where-live.py
class Person:
    def __init__(self, name, bunji, region):
        self.name = name
        self.bunji = bunji
        self.region = region

n = int(input())
arr = [tuple(input().split()) for _ in range(n)]
people = [Person(name, bunji, region) for name, bunji, region in arr]

target_idx = 0
for idx, val in enumerate(people):
    if val.name > people[target_idx].name:
        target_idx = idx

print(f'name {people[target_idx].name}')
print(f'addr {people[target_idx].bunji}')
print(f'city {people[target_idx].region}')